schemas/publication.json

```diff
@@ -46,13 +46,6 @@
 			"apiSummary": true,
 			"readOnly": true
 		},
-		"categoryIds": {
-			"type": "array",
-			"apiSummary": true,
-			"items": {
-				"type": "integer"
-			}
-		},
```

Comment:
Thanks @jonasraoni! This is a valid concern you pointed out given the complexity, but i removed `categoryIds` from this schema after discussion with @asmecher to optimize the handling of category data in a manner consistent with our updated architecture and isn't stored redundantly. I’m also using  transitional approach to access `categoryIds` in memory without saving them directly to maintain compatibility while moving towards a more efficient data model. Here's the transitional approach
```
protected function setCategories(Publication $publication): void {
    $categoryIds = PublicationCategory::withPublicationId($publication->getId())->pluck('category_id');
    $publication->setData('categoryIds', $categoryIds);
}
```
This method ensures that categoryIds are dynamically accessible in-memory for those parts of our codebase that rely on them, without storing them directly in the publication's properties.

Also about setting the readOnly attribute to true, yes this could be a potential solution, but removing the field entirely avoids confusion and encourage the use of our new data model.

I think it would be good to discuss further and have your insights on this transition @jonasraoni and @asmecher

Reply:
@Hafsa-Naeem my only concern here is regarding the intertwined code. This "JSON" model might have a lot of `readOnly` attributes, such attributes are not going to be stored, but they are used to generate the API documentation and perhaps in other places.

Did you dig deeper on this and watched if the APIs which retrieve publications are working properly? You might also watch the output of `tools/buildSwagger.php` with and without this change, which is the API documentation.